les/server_handler.go

```diff
@@ -223,648 +221,109 @@ func (h *serverHandler) handleMsg(p *clientPeer, wg *sync.WaitGroup) error {
 	}
 	defer msg.Discard()
 
-	var (
-		maxCost uint64
-		task    *servingTask
-	)
 	p.responseCount++
```

Comment:
Is it problematic? The description to `responseLock` is 

```
// responseLock ensures that responses are queued in the same order as
// RequestProcessed is called
```

while the responseCount here is bumped without lock protection.

Reply:
`handleMsg` is called sequentially so it does not need to be thread safe and this is the only place where `p.responseCount` is accessed. The lock is used in the serving goroutine and what it really ensures is that `RequestProcessed` and `queueSend` calls are done in the same order. So I think this one is ok (it works the same way in the original version).

Reply:
Right. Resolved it.